typings/index.d.ts

```diff
@@ -244,6 +261,16 @@ declare module 'discord.js' {
     public setRTCRegion(region: string | null): Promise<this>;
   }
 
+  export class BaseMessageComponent {
+    constructor(data?: BaseMessageComponent | BaseMessageComponentOptions);
+    public type: MessageComponentType | null;
+    public setType(type: MessageComponentType | number): this;
+  }
+
+  interface BaseMessageComponentOptions {
+    type?: MessageComponentType;
```

Comment:
Related to my above suggestion:
```suggestion
    type?: MessageComponentType | MessageComponentTypes;
```
(This also should be moved below to the other interfaces.)